src/test/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcherTest.java

```diff
@@ -0,0 +1,110 @@
+    @MethodSource("performSearchParameters")
+    public void performSearchOnEmptyQuery(Set<SearchBasedFetcher> fetchers) {
+        CompositeSearchBasedFetcher compositeFetcher = new CompositeSearchBasedFetcher(fetchers);
+
+        List<BibEntry> queryResult = compositeFetcher.performSearch("");
+
+        Assertions.assertTrue(queryResult.isEmpty());
+    }
+
+    @ParameterizedTest(name = "Perform search on query \"quantum\". Using the CompositeFetcher of the following " +
+            "Fetchers: {arguments}")
+    @MethodSource("performSearchParameters")
+    public void performSearchOnNonEmptyQuery(Set<SearchBasedFetcher> fetchers) {
+        CompositeSearchBasedFetcher compositeFetcher = new CompositeSearchBasedFetcher(fetchers);
+
+        List<BibEntry> compositeResult = compositeFetcher.performSearch("quantum");
+
+        for (SearchBasedFetcher fetcher : fetchers) {
+            try {
+                Assertions.assertTrue(compositeResult.containsAll(fetcher.performSearch("quantum")));
+            } catch (FetcherException e) {
+                /* We catch the Fetcher exception here, since the failing fetcher also fails in the CompositeFetcher
+                 * and just leads to no additional results in the returned list. Therefor the test should not fail
+                 * due to the fetcher exception
+                 */
+                LOGGER.warn(String.format("Fetcher %s failed ", fetcher.getName()), e);
+            }
+        }
+    }
+
+    /**
+     * This method provides other methods with different sized sets of Searchbased fetchers wrapped in Arguments.
+     *
+     * @return A stream of Arguments wrapping the sets.
+     */
+    static Stream<Arguments> performSearchParameters() {
+        ImportFormatPreferences importFormatPreferences = mock(ImportFormatPreferences.class);
+        when(importFormatPreferences.getFieldContentFormatterPreferences())
+                .thenReturn(mock(FieldContentFormatterPreferences.class));
+        List<Set<SearchBasedFetcher>> fetcherParameters = new ArrayList<>();
+        List<SearchBasedFetcher> list = new ArrayList<>();
+
+        list.add(new ArXiv(importFormatPreferences));
+        list.add(new INSPIREFetcher(importFormatPreferences));
+        list.add(new GvkFetcher());
+        list.add(new AstrophysicsDataSystem(importFormatPreferences));
+        list.add(new MathSciNet(importFormatPreferences));
+        list.add(new ZbMATH(importFormatPreferences));
+        list.add(new GoogleScholar(importFormatPreferences));
+        list.add(new DBLPFetcher(importFormatPreferences));
+        list.add(new SpringerFetcher());
+        list.add(new CrossRef());
+        list.add(new CiteSeer());
+        list.add(new DOAJFetcher(importFormatPreferences));
+        list.add(new IEEE(importFormatPreferences));
+        /* Disabled due to issue regarding Comparison: Title fields of the entries that otherwise are equivalent differ
+         * due to different JAXBElements.
```

Comment:
Can you provide me the concreete example? Maybe, we need to fix the MedlineFetcher?

Reply:
AFAIK:
When using the same instance of MedlineFetcher to fetch results directly or as part of the CompositeFetcher, certain BibEntries have differing titles due to something related to JAXBElements.
When searching for "Indistinguishable Photons from Deterministically" the direct use of the MedLine Fetcher returns a BibEntry with the title:

> title = {Indistinguishable Photons from Deterministically Integrated Single Quantum Dots in Heterogeneous GaAs/Si, javax.xml.bind.JAXBElement@31b82e0f, N, javax.xml.bind.JAXBElement@27a09971, Quantum Photonic Circuits.}

The CompositeFetcher, on the other hand, returns a BibEntry with the title:

> title = {Indistinguishable Photons from Deterministically Integrated Single Quantum Dots in Heterogeneous GaAs/Si, javax.xml.bind.JAXBElement@3289079a, N, javax.xml.bind.JAXBElement@32fa809f, Quantum Photonic Circuits.} 

Both would be the same only differing by the JAXBElement Instance.